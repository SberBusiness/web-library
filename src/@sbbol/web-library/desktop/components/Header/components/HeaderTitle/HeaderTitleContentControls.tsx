import * as React from 'react';
import {classnames} from '@sbbol/web-library/common/utils/classnames/classnames';

/**
 * Свойства компонента.
 */
export interface IHeaderTitleContentControlsProps extends React.HTMLAttributes<HTMLDivElement> {}

/**
 * Блок с кнопками действий HeaderTitle.
 */
export const HeaderTitleContentControls: React.FC<IHeaderTitleContentControlsProps> = ({children, className, ...htmlDivAttributes}) => (
    <div className={classnames(className, 'cssClass[globalHeaderTitleContentControls]')} {...htmlDivAttributes}>
        {children}
    </div>
);

HeaderTitleContentControls.displayName = 'HeaderTitleContentControls';
