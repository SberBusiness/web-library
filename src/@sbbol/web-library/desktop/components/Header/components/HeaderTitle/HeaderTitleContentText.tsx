import * as React from 'react';
import {classnames} from '@sbbol/web-library/common/utils/classnames/classnames';

interface IHeaderTitleContentTextProps extends React.HTMLAttributes<HTMLHeadingElement> {}

/**
 * Основной заголовок.
 */
export class HeaderTitleContentText extends React.PureComponent<IHeaderTitleContentTextProps> {
    public static displayName = 'HeaderTitleContentText';

    public render(): React.ReactNode {
        const {children, className, ...HTMLHeadingAttributes} = this.props;

        return (
            <h1 className={classnames(className, 'cssClass[globalHeaderTitleContentText]')} {...HTMLHeadingAttributes}>
                {children}
            </h1>
        );
    }
}
