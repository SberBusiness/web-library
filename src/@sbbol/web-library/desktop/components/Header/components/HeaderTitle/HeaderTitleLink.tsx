import * as React from 'react';
import {ELinkSize, ELinkType, ILinkTextProps, Link} from '../../../Link/Link';

interface IHeaderTitleLinkProps extends Omit<ILinkTextProps, 'linkType' | 'size'> {}

/**
 * Ссылка. Может вести к началу сценария/раздела.
 */
export class HeaderTitleLink extends React.PureComponent<IHeaderTitleLinkProps> {
    public static displayName = 'HeaderTitleLink';

    public render(): React.ReactNode {
        const {children, ...linkProps} = this.props;

        return (
            <div className="cssClass[headerTopLinkWrapper]">
                <Link {...linkProps} linkType={ELinkType.TEXT} size={ELinkSize.LG}>
                    {children}
                </Link>
            </div>
        );
    }
}
