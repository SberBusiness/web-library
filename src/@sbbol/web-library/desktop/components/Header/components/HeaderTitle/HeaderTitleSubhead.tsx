import * as React from 'react';
import {classnames} from '@sbbol/web-library/common/utils/classnames/classnames';

/**
 * Свойства компонента.
 */
export interface IHeaderTitleSubheadProps extends React.HTMLAttributes<HTMLDivElement> {}

/**
 * Подзаголовок.
 */
export const HeaderTitleSubhead: React.FC<IHeaderTitleSubheadProps> = ({children, className, ...htmlDivAttributes}) => (
    <div className={classnames(className, 'cssClass[headerTitleSubhead]')} {...htmlDivAttributes}>
        {children}
    </div>
);

HeaderTitleSubhead.displayName = 'HeaderTitleSubhead';
