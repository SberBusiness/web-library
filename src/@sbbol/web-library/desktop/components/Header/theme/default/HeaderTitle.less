@import '../../../../common/theme/default';

.globalHeaderTitle {
    padding: @ufs-theme-page-padding-y @ufs-theme-page-padding-x;

    .headerTitleLinkWrapper {
        .font-size(@font-size-14);
    }

    .headerTitleContent {
        display: flex;
        justify-content: space-between;

        .globalHeaderTitleContentText {
            .h1-header-main();

            box-sizing: border-box;
            text-transform: initial;
        }

        .globalHeaderTitleContentControls {
            white-space: nowrap;
            padding-left: @ufs-theme-padding-before-controls-column;
        }

        // На мобильных устройствах блок с кнопками располагается строкой ниже.
        @media (max-width: @screen-md) {
            flex-wrap: wrap;

            .globalHeaderTitleContentText {
                width: 100%;
            }

            .globalHeaderTitleContentControls {
                padding-top: @page-elements-padding-y-sm;
                padding-left: 0;
                white-space: normal;
            }
        }
    }

    .headerTitleSubhead {
        .font-size(@font-size-14);

        color: @color-asphalt-01;

        // На мобильных устройствах добавляется отступ от верхнего блока с кнопками.
        @media (max-width: @screen-md) {
            padding-top: @page-elements-padding-y-sm;
        }
    }
}

@media (max-width: @screen-md) {
    .globalHeaderTitle {
        padding: @page-padding-sm;

        .headerTitleContent .globalHeaderTitleContentText {
            .font-size(@font-size-16);
        }
    }
}
