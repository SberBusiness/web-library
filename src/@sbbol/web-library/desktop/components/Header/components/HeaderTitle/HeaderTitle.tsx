import {HeaderTitleContent} from '@sbbol/web-library/desktop/components/Header/components/HeaderTitle/HeaderTitleContent';
import {HeaderTitleLink} from '@sbbol/web-library/desktop/components/Header/components/HeaderTitle/HeaderTitleLink';
import {HeaderTitleSubhead} from '@sbbol/web-library/desktop/components/Header/components/HeaderTitle/HeaderTitleSubhead';
import {classnames} from '@sbbol/web-library/common/utils/classnames/classnames';
import * as React from 'react';

interface IHeaderTitleProps extends React.HTMLAttributes<HTMLDivElement> {}

/**
 * Первый уровень Header. Содержит ссылку назад, заголовок, кнопки действий и подзаголовок.
 */
export class HeaderTitle extends React.Component<IHeaderTitleProps> {
    public static displayName = 'HeaderTitle';

    public static Link = HeaderTitleLink;
    public static Content = HeaderTitleContent;
    public static Subhead = HeaderTitleSubhead;

    public render(): React.ReactNode {
        const {children, className, ...htmlDivAttributes} = this.props;
        return (
            <div className={classnames(className, 'cssClass[globalHeaderTitle]')} {...htmlDivAttributes}>
                {children}
            </div>
        );
    }
}
