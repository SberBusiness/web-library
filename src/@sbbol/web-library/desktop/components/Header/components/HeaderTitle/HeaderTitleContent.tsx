import {HeaderTitleContentControls} from '@sbbol/web-library/desktop/components/Header/components/HeaderTitle/HeaderTitleContentControls';
import {HeaderTitleContentText} from '@sbbol/web-library/desktop/components/Header/components/HeaderTitle/HeaderTitleContentText';
import * as React from 'react';
import {classnames} from '@sbbol/web-library/common/utils/classnames/classnames';

interface IHeaderTitleContentProps extends React.HTMLAttributes<HTMLDivElement> {}

/**
 * Часть HeaderTitle с заголовокм и кнопками действий.
 */
export class HeaderTitleContent extends React.PureComponent<IHeaderTitleContentProps> {
    public static displayName = 'HeaderTitleContent';

    public static Text = HeaderTitleContentText;
    public static Controls = HeaderTitleContentControls;

    public render(): React.ReactNode {
        const {children, className, ...htmlDivAttributes} = this.props;

        return (
            <div className={classnames(className, 'cssClass[headerTitleContent]')} {...htmlDivAttributes}>
                {children}
            </div>
        );
    }
}
