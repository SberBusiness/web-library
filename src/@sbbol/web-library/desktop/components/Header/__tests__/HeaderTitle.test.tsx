import React from 'react';
import renderer from 'react-test-renderer';
import {HeaderTitle} from '../components/HeaderTitle/HeaderTitle';
import {allure} from '@jest/unit/allure-report';

describe('HeaderTitle', () => {
    beforeEach(() => {
        allure.feature('HeaderTitle');
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <HeaderTitle>
                    <HeaderTitle.Link>Link text</HeaderTitle.Link>
                    <HeaderTitle.Content>
                        <HeaderTitle.Content.Text>Text</HeaderTitle.Content.Text>
                        <HeaderTitle.Content.Controls>Controls</HeaderTitle.Content.Controls>
                    </HeaderTitle.Content>
                    <HeaderTitle.Subhead>Text</HeaderTitle.Subhead>
                </HeaderTitle>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
