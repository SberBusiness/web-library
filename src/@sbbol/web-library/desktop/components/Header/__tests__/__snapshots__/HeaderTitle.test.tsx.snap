// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle renders correctly 1`] = `
<div
  className="cssClass[globalHeaderTitle]"
>
  <div
    className="cssClass[headerTopLinkWrapper]"
  >
    <a
      className="cssClass[link] hoverable cssClass[lg] cssClass[text] cssClass[focusVisible]"
      onBlur={[Function]}
      onMouseDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Link text
    </a>
  </div>
  <div
    className="cssClass[headerTitleContent]"
  >
    <h1
      className="cssClass[globalHeaderTitleContentText]"
    >
      Text
    </h1>
    <div
      className="cssClass[globalHeaderTitleContentControls]"
    >
      Controls
    </div>
  </div>
  <div
    className="cssClass[headerTitleSubhead]"
  >
    Text
  </div>
</div>
`;
